### DISCUSSION TITLE
log(1+inv_pos(x)) error

### DISCUSSION THREAD

**Discussion by blyspyder** [yalmip/YALMIP#1143] [Problems using YALMIP]
K>> pj = sdpvar(1,4)
Linear matrix variable 1x4 (full, real, 4 variables)
Coeffiecient range: 1 to 1
K>> repmat(pj,4,1)
Linear matrix variable 4x4 (full, real, 4 variables)
Coeffiecient range: 1 to 1
K>> log2(1+ repmat(pj,4,1) .* inv_pos( repmat(pj,4,1)))
Nonlinear matrix variable 4x1 (full, real, 4 variables)
Coeffiecient range: 1.4427 to 1.4427

I want the shape of output "log2(1+ repmat(pj,4,1) .* inv_pos( repmat(pj,4,1)))" is (4,4), but it is (4,1). 
I don't know what happen? How can I get a result with shape (4,4)

**Comment by johanlofberg** [comment]
You are mixing yalmip with cvx operator inv_pos. YaLMIP has a similar called invpos

```
y=log2(1+repmat(pj,4,1).* invpos( repmat(pj,4,1)));
size(y)

ans =

     4     4
```

However, this will never work as you are multiplying convexity-aware operators, and that will not pass the modelling layers

If you simply want to solve problem using a general nonlinear solver, just use 1./pj, and if you want to use an exponential cone solver, you will have to write the expression using the supported operators and possibly extra variables and epi/hypo-graphs etc https://yalmip.github.io/tutorial/exponentialcone/
